python/ray/tests/test_multi_node.py

```diff
@@ -167,6 +167,39 @@ def f():
         assert "success" in out
 
 
+def test_cleanup_on_driver_exit(call_ray_start):
```

Comment:
Adding this test to replace `test_monitors.py` (which wasn't even being run anyway).